public/app/features/browse-dashboards/BrowseDashboardsPage.tsx

```diff
@@ -31,13 +32,23 @@ export interface Props extends GrafanaRouteComponentProps<BrowseDashboardsPageRo
 
 const BrowseDashboardsPage = memo(({ match }: Props) => {
   const { uid: folderUID } = match.params;
+  const dispatch = useDispatch();
 
   const styles = useStyles2(getStyles);
   const [searchState, stateManager] = useSearchStateManager();
   const isSearching = stateManager.hasSearchFilters();
 
   useEffect(() => stateManager.initStateFromUrl(folderUID), [folderUID, stateManager]);
 
+  // Clear selected state when folderUID changes
+  useEffect(() => {
+    dispatch(
+      setAllSelection({
+        isSelected: false,
+      })
+    );
+  }, [dispatch, folderUID]);
```

Comment:
I think this can move to the `useEffect` where `stateManager.initStateFromUrl` is?